src/tests/save_component/save_comp_helper.rs

```diff
@@ -17,30 +20,38 @@ pub mod save_comp_helper {
         let str_query = format!("get-component: {} save-as test", system);
         let query = parse_queries::parse_to_expression_tree(str_query.as_str()).remove(0);
 
-        let mut clock_index: u32 = 0;
-        let base_system = if let QueryExpression::GetComponent(expr) = &query {
+        let mut dim: u32 = 0;
+        let (base_system, new_system) = if let QueryExpression::GetComponent(expr) = &query {
             let mut comp_loader = project_loader.to_comp_loader();
-            extract_system_rep::extract_side(expr.as_ref(), &mut *comp_loader, &mut clock_index)
+            (
+                extract_system_rep::get_system_recipe(expr.as_ref(), &mut *comp_loader, &mut dim),
+                extract_system_rep::get_system_recipe(expr.as_ref(), &mut *comp_loader, &mut dim),
+            )
         } else {
             panic!("Failed to create system")
         };
+        //dim += 2;
 
-        let mut new_comp = combine_components(&base_system.clone());
-        new_comp.create_edge_io_split();
-        let mut new_comp = Box::new(new_comp);
-        decl.add_component(&new_comp);
+        let new_comp = new_system.compile(dim);
+
+        if let Err(_) = new_comp {
+            return;
+        }
```

Comment:
I feel like the component failing to compile should be considered a failing test, other wise it looks like it succeeds and refines itself.

Reply:
This was mostly done because our save component tests are just random combinations of operators and components so we don't really know whether they are in fact valid so they are not expected to compile without error e.g. many of the made up conjunctions have mismatched actions or are not least consistent. 
We could instead remove all the test cases that fail due to non-compiling components, if you prefer that?

Reply:
Good reasoning!